### PR TITLE
feat: introduce aws-ecs byovpc sandbox

### PR DESCRIPTION
**Pull request by jonmorehouse**
This PR introduces a built-in sandbox for `aws-ecs-byovpc`. This sandbox uses inputs to wire in existing vpcs, and creates a fargate cluster and the necessary IAM roles for our a.) runner installer b.) the static runner and c.) the odr.

Notably, the runner installer is in Nuon's infrastructure and is the role that we will assume when doing the installation. The static runner _only_ creates/deletes the ODRs, and hte ODR is the thing that needs the most permissions, as it manages the components.

### CODE REVIEW

File: aws-ecs-byovpc/error-destroy.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+set -u
+set -o pipefail
+
+echo "executing error-destroy script"
+
+echo "ensuring AWS is setup"
+aws sts get-caller-identity > /dev/null
+
+echo "looking for ENIs which were orphaned by vpc-cni plugin"
```
Comment (jordan-acosta) [inline aws-ecs-byovpc/error-destroy.sh:11]: I thought this was just for EKS?

File: aws-ecs-byovpc/example.tfvars [hunk 2]
```diff
@@ -0,0 +1,14 @@
+nuon_id = "28g232g0an6vh29t6mu7kv3a96"
+region  = "us-east-2"
+# assume_role_arn  = "arn:aws:iam::949309607565:role/nuon-demo-install-access"
+assume_role_arn  = "" # This needs to be set for use from our services but should not be when running locally.
+install_role_arn = "arn:aws:iam::618886478608:role/install-k8s-admin-stage"
```
Comment (jordan-acosta) [inline aws-ecs-byovpc/example.tfvars:5]: It looks like some EKS stuff got left here.

File: aws-ecs-byovpc/variables.tf [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+variable "cluster_name" {
+  type        = string
+  description = "The name of the EKS cluster. Will use the install ID by default."
+}
+
+variable "eks_version" {
+  type        = string
+  description = "The Kubernetes version to use for the EKS cluster."
+  default     = ""
+}
```
Comment (jordan-acosta) [inline aws-ecs-byovpc/variables.tf:62]: I assume there are other config values that would be relevant to ECS?

**Review by jordan-acosta — COMMENTED** [review]
I found a few issues, but it looks like it's mostly stuff left over from copy/pasting the EKS sandbox.

File: aws-ecs-byovpc/error-destroy.sh (hunk 1 above)
Comment (jonmorehouse) [reply aws-ecs-byovpc/error-destroy.sh:11]: @jordan-acosta leaving this cleanup in here, for now so I can monitor from a known good, and then if we see on deprovision that no ENIs are leaking, we can go ahead and remove this.

File: aws-ecs-byovpc/example.tfvars (hunk 2 above)
Comment (jonmorehouse) [reply aws-ecs-byovpc/example.tfvars:5]: Yep, cleaning up. Thanks!

**Pull request merged**